backend/app/rest/api/rest_private_test.go

```diff
@@ -450,6 +450,47 @@ func TestRest_Vote(t *testing.T) {
 	assert.Equal(t, map[string]bool(nil), cr.Votes)
 }
 
+func TestRest_Email(t *testing.T) {
+	ts, _, teardown := startupT(t)
+	defer teardown()
+
+	const goodToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJyZW1hcms0MiIsImV4cCI6MTU3MTY5MjU2MSwiaXNzIjoicmVtYXJrNDIiL" +
+		"CJuYmYiOjE1NzE2OTA3MDEsImhhbmRzaGFrZSI6eyJpZCI6ImRldjo6Z29vZEBleGFtcGxlLmNvbSJ9fQ.QQ-Q2PbXYnKDXPMSfcpNjoh8xzh1zYdL_F_Xgy7czZc"
+
+	var testData = []struct {
+		description  string
+		url          string
+		method       string
+		responseCode int
+		noAuth       bool
+	}{
+		{description: "issue delete request without auth", url: "/api/v1/email", method: http.MethodDelete, responseCode: http.StatusUnauthorized, noAuth: true},
+		{description: "issue delete request without site_id", url: "/api/v1/email", method: http.MethodDelete, responseCode: http.StatusBadRequest},
+		{description: "delete non-existent user email", url: "/api/v1/email?site=remark42", method: http.MethodDelete, responseCode: http.StatusOK},
+		{description: "set user email, token not set", url: "/api/v1/email?site=remark42", method: http.MethodPut, responseCode: http.StatusBadRequest},
+		{description: "send confirmation without address", url: "/api/v1/email?site=remark42", method: http.MethodGet, responseCode: http.StatusBadRequest},
+		{description: "send confirmation", url: "/api/v1/email?site=remark42&address=good@example.com", method: http.MethodGet, responseCode: http.StatusOK},
+		{description: "set user email, token is good", url: fmt.Sprintf("/api/v1/email?site=remark42&tkn=%s", goodToken), method: http.MethodPut, responseCode: http.StatusOK},
```

Comment:
I don't test for claims token's User.Email update there and would implement such a test if necessary.